src/backend/core/plugins/base.py

```diff
@@ -11,3 +11,9 @@ class BaseOrganizationPlugin:
     def run_after_create(self, organization) -> None:
         """Method called after creating an organization."""
         raise NotImplementedError("Plugins must implement the run_after_create method")
+
+    def run_after_grant_access(self, organization_access) -> None:
+        """Method called after creating an organization."""
+        raise NotImplementedError(
+            "Plugins must implement the run_after_grant_access method"
```

Comment:
Should these plugin methods really be mandatory?

Reply:
In this PR the intent is to extend the plugin mechanism (introduced in #584) remaining consistent with prior choices, among which was this abstract base class behaviour. This is common in Python and could have other rationales, such as protecting against non-intuitive method dispatching, or ensuring plugin implementors take care to consider the entire life cycle when creating a plugin.

We can loop in @qbey ?